include/lsst/meas/astrom/matchOptimisticB.h

```diff
@@ -0,0 +1,137 @@
+// -*- lsst-c++ -*-
+#if !defined(LSST_MEAS_ASTROM_MATCHOPTIMISTICB_H)
+#define LSST_MEAS_ASTROM_MATCHOPTIMISTICB_H
+
+#include <cmath>
+#include <string>
+#include <vector>
+
+#include "lsst/pex/config.h"
+#include "lsst/afw/table/Source.h"
+#include "lsst/afw/table/Match.h"
+
+namespace lsst {
+namespace meas {
+namespace astrom {
+
+    /**
+    A wrapper around a SimpleRecord that allows us to associate an pixel position
+    with a position reference star.
+
+    RecordProxy can also wrap a SourceRecord (in which case the pixel position in the proxy
+    should match that in the record).
+    */
+    struct RecordProxy {
+        PTR(lsst::afw::table::SimpleRecord) record;
+        lsst::afw::geom::Point2D position;
```

Comment:
I decided it was a reasonable construct so I left it alone. It allows the code to easily work on pairs of any combination of ref objects and sources, which is something it needs to do.

But I did mean to make it internal to the .cc file.